Sources/CacheAdvance/CacheAdvance.swift

```diff
@@ -30,8 +30,8 @@ public final class CacheAdvance<T: Codable> {
     ///   - maximumBytes: The maximum size of the cache, in bytes. Logs larger than this size will fail to append to the store.
     ///   - shouldOverwriteOldMessages: When `true`, once the on-disk store exceeds maximumBytes, new entries will replace the oldest entry.
     ///
-    /// - Warning: `maximumBytes` must be consistent for the life of a cache. Changing this value after logs have been persisted to a cache leads to undefined behavior.
-    /// - Warning: `shouldOverwriteOldMessages` must be consistent for the life of a cache. Changing this value after logs have been persisted to a cache leads to undefined behavior.
+    /// - Warning: `maximumBytes` must be consistent for the life of a cache. Changing this value after logs have been persisted to a cache will lead to data loss.
+    /// - Warning: `shouldOverwriteOldMessages` must be consistent for the life of a cache. Changing this value after logs have been persisted to a cache will lead to data loss.
```

Comment:
I need to think on this one. I can't just assert here because that'd break tests. Throwing and making the caller take care of it seems like a reasonable option. Will sit on this one (and come to a conclusion before merging)

Reply:
The bummer about throwing is that then everyone who uses this needs to use a `do {} catch {}` dance. Maybe a middle ground would be providing a throwable and non-throwable version. I think it should be _possible_ to know that you are trying to open a cache file that has parameters that don’t match what you expect. Then if you want to be safe, you can avoid opening it (and avoid losing all of your data). But I also see value in letting someone open it easily, discarding data if necessary. After all it is a cache...

Reply:
We already can `throw` when we open the file, but not in a way that is terribly recoverable.

One thing is that I don't want to actually read data from this file in `init`, since `init` should be cheap. At which point we'd be validating our header and throwing this error on `append(message:)` or `messages()`. Maybe I could offer a `validateHeader` method that can be called?

Part of the reason I punted on surfacing this error is there are only two ways I can think of to hit it today:

1. Someone points a CacheAdvance at a file that already exists and has been written to by another object. This is very clearly user error, and not something that another persistence system would necessarily surface.
1. Someone updates from a `0.0.3` cache to a `0.1.0` cache. This doesn't seem likely since this repo is very much in beta, and current traffic does not suggest that anyone is using this repo today.

When we eventually add a new breaking header change, I think we'll need to think about this problem more closely. But I'm not convinced this is something we need to handle in this release. But it is certainly something we'll need to handle better in the future.

Reply:
That all makes sense.
I think a `validateHeader` method would be a good addition.
I like libraries that give you the tools to be safe. You don’t always need to use them, but it’s nice to know that they’re there.

Reply:
(Not urgent though)

Reply:
I’ll file an issue on this one. Seems like a reasonable addition to get to.